cloud/misc/rhevm.py

```diff
@@ -0,0 +1,1476 @@
+
+DOCUMENTATION = '''
+---
+module: rhevm
+author: Timothy Vandenbrande
+short_description: RHEV/oVirt automation
+description:
+    - Allows you to create/remove/update or powermanage virtual machines on a RHEV/oVirt platform.
+version_added: "2.2"
+requirements:
+    - ovirtsdk
+options:
+    user:
+        description:
+            - The user to authenticate with.
+        default: "admin@internal"
+        required: false
+    server:
+        description:
+            - The name/ip of your RHEV-m/oVirt instance.
+        default: "127.0.0.1"
+        required: false
+    port:
+        description:
+            - The port on which the API is reacheable.
+        default: "443"
+        required: false
+    insecure_api:
+        description:
+            - A boolean switch to make a secure or insecure connection to the server.
+        default: false
+        required: false
+    cluster:
+        description:
+            - The rhev/ovirt cluster in which you want you VM to start.
+        required: true
+    state:
+        description:
+            - This serves to create/remove/update or powermanage your VM.
+        default: "present"
+        required: false
+        choices: ['present', 'absent', 'up', 'down', 'restart', 'cd']
+    image:
+        description:
+            - The template to use for the VM.
+        default: null
+        required: false
+    type:
+        description:
+            - To define if the VM is a server or desktop.
+        default: server
+        required: false
+        choices: [ 'server', 'desktop', 'host' ]
+    vmcpu:
+        description:
+            - The number of cpu's you want in your VM.
+        default: "2"
+        required: false
+    vmmem:
+        description:
+            - The amount of memory you want your VM to use (in GB).
+        default: "1"
+        required: false
+    osver:
+        description:
+            - The operationsystem option in RHEV/oVirt.
+        default: "rhel_6x64"
+        required: false
+    mempol:
+        description:
+            - The minimum amount of memory you wish to reserve for this system.
+        default: "1"
+        required: false
+    vm_ha:
+        description:
+            - To make you VM High Available.
+        default: true
+        required: false
+    disks:
+        description:
+            - This option uses complex arguments and is a list of disks with the options name, size and domain.
+        required: false
+    nics:
+        description:
+            - This option uses complex arguments and is a list of nics with the options name and vlan.
+        required: false
+    boot_order:
+        description:
+            - This option uses complex arguments and is a list of items that specify the bootorder.
+        default: ["network","hd"]
+        required: false
+    cd_drive:
+        description:
+            - The CD you wish to have mounted on the VM when state = 'CD'.

```

Comment:
To make this a bit more readable you can do:

`I(state = 'CD')`

From http://docs.ansible.com/ansible/developing_modules.html#module-checklist
Ensure that you make use of U() for urls, C() for files and options, I() for params, M() for modules.

See http://docs.ansible.com/ansible/cloudflare_dns_module.html for an example of this